clients/filesystem-fuse/src/filesystem.rs

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+use fuse3::{Errno, FileType, Timestamp};
+use std::time::SystemTime;
+
+pub type Result<T> = std::result::Result<T, Errno>;
+
+/// File system interface for the file system implementation. it use by FuseApiHandle
+/// the `file_id` and `parent_file_id` it is the unique identifier for the file system, it is used to identify the file or directory
+/// the `fh` it is the file handle, it is used to identify the opened file, it is used to read or write the file content
+pub trait IFileSystem: Send + Sync {
+    fn get_file_path(&self, file_id: u64) -> String;
```

Comment:
add async to all interfaces?

Reply:
add Result to wrap errors for all interfaces?

Reply:
We'll consider adding async later.